packages/grid/x-data-grid/src/components/cell/GridEditDateCell.tsx

```diff
@@ -98,24 +98,32 @@ function GridEditDateCell(props: GridEditDateCellProps) {
   const ownerState = { classes: rootProps.classes };
   const classes = useUtilityClasses(ownerState);
 
+  const hasUpdatedValueOnRender = React.useRef(false);
```

Comment:
```suggestion
  const hasUpdatedEditValueOnMount = React.useRef(false);
```
Would this name better reflect what it's used for?